legend-engine-xts-authentication/legend-engine-xt-authentication-implementation-core/src/main/java/org/finos/legend/connection/ConnectionFactory.java

```diff
@@ -0,0 +1,44 @@
+// Copyright 2023 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.connection;
+
+import org.finos.legend.authentication.credentialprovider.CredentialBuilder;
+import org.finos.legend.authentication.credentialprovider.CredentialProviderProvider;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.authentication.specification.AuthenticationSpecification;
+import org.finos.legend.engine.shared.core.identity.Credential;
+import org.finos.legend.engine.shared.core.identity.Identity;
+
+public class ConnectionFactory
+{
+    private final ConnectionSetupFlowProvider flowProviderHolder;
+    private final CredentialProviderProvider credentialProviderProvider;
+
+    public ConnectionFactory(ConnectionSetupFlowProvider flowProviderHolder, CredentialProviderProvider credentialProviderProvider)
```

Comment:
ConnectionSetup is an odd name. Maybe ConnectionSpeficiationFlowProvider ?

Reply:
This renaming is done, but it'll take some time for me to unify this and the old `ConnectionSpecification` logic since I need to test the flows which are currently used by `ServiceStore` and `MongoDB`